src/tiled/automapper.cpp

```diff
@@ -712,8 +712,18 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 bool matchListYes = false;
                 bool matchListNo  = false;
 
-                if (!setLayer->contains(x + offset.x(), y + offset.y()))
-                    return false;
+
+                if (!setLayer->contains(x + offset.x(), y + offset.y())) {
+                    foreach (const TileLayer *comparedTileLayer, listYes) {
+                        if (!comparedTileLayer->contains(x, y))
+                            return false;
```

Comment:
I understand that rather than discarding the whole match just because the output layer does not contain every position in the rule region, you're instead taking a shortcut in only checking the `listYes`, since the `listNo` is irrelevant for determining an out-of-range match?

What I don't understand yet, is why when a layer in `listYes` does not contain this rule region position, should the match fail? Shouldn't that be treated like the cell being empty, thus doing `continue`?

In general, wouldn't it make sense to make sure the layers in `listYes` and `listNo` are all the same size, and all big enough to contain the `ruleRegion`? Then we can drop all the `contains` checks on them. Or is there some reason why this is not the case?